xarray/backends/api.py

```diff
@@ -562,12 +566,26 @@ def open_mfdataset(paths, chunks=None, concat_dim=_CONCAT_DIM_DEFAULT,
 
     if lock is None:
         lock = _default_lock(paths[0], engine)
-    datasets = [open_dataset(p, engine=engine, chunks=chunks or {}, lock=lock,
-                             **kwargs) for p in paths]
-    file_objs = [ds._file_obj for ds in datasets]
 
-    if preprocess is not None:
-        datasets = [preprocess(ds) for ds in datasets]
+    open_kwargs = dict(engine=engine, chunks=chunks or {}, lock=lock,
+                       autoclose=autoclose, **kwargs)
+
+    if parallel or (parallel is None and get_scheduler() == 'distributed'):
+        import dask
+        datasets = [delayed(open_dataset)(p, **open_kwargs) for p in paths]
+        # important: get the file_objs before calling preprocess
+        file_objs = [ds._file_obj for ds in datasets]
+        if preprocess is not None:
+            datasets = [delayed(preprocess)(ds) for p in datasets]
+
+        # calling compute here will return compute the datasets/file_objs lists
+        # the underlying datasets will still be stored as dask arrays
+        dask.compute([datasets, file_objs])
```

Comment:
Don't you need to assign these computed results, e.g., `datasets, file_objs = dask.compute([datasets, file_objs])`?